src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -1224,7 +1224,7 @@ public void CheckAndMarkAsPageable(AsNode node)
                 FlagPathAsAsync(node);
 
                 // Pageable nodes are also stateful as data is always pulled from outside.
-                SetStateful(node, isStateful: true);
+                SetStateful(node.Left, isStateful: true);
```

Comment:
>Left [](http://example.com/codeflow?start=33&length=4)

This is a significant change. Can you elaborate on why this is needed?

Reply:
I updated PR description and this fix